tensorflow/compiler/xla/tools/replay_computation.cc

```diff
@@ -400,21 +400,31 @@ StatusOr<HloSnapshot> ParseSingleHloFile(const string& filename,
           filename.c_str());
 
   if (tensorflow::ReadBinaryProto(env, filename, snapshot.mutable_hlo()).ok()) {
-    return snapshot;
+    return std::vector<HloSnapshot>{std::move(snapshot)};
   }
   fprintf(stderr, "%s: is not HloProto. Trying HLO text.\n", filename.c_str());
   string contents;
   TF_RETURN_IF_ERROR(tensorflow::ReadFileToString(env, filename, &contents));
   HloModuleConfig config;
   config.set_debug_options(GetDebugOptionsFromFlags());
-  StatusOr<std::unique_ptr<HloModule>> module =
-      ParseAndReturnUnverifiedModule(contents, config);
-  if (module.ok()) {
-    *snapshot.mutable_hlo()->mutable_hlo_module() =
+  std::vector<std::string> hlo_module_texts =
+    absl::StrSplit(contents, "// -----");
+  std::vector<HloSnapshot> snapshots;
+  for (const std::string& hlo_module_text : hlo_module_texts) {
+    StatusOr<std::unique_ptr<HloModule>> module =
+      ParseAndReturnUnverifiedModule(hlo_module_text, config);
+    if (module.ok()) {
+      HloSnapshot snapshot;
+      *snapshot.mutable_hlo()->mutable_hlo_module() =
         module.ValueOrDie()->ToProto();
-    return snapshot;
-  } else {
-    LOG(ERROR) << module.status();
+      //return std::vector<HloSnapshot>{std::move(snapshot)};
```

Comment:
Stray comment?

Reply:
I amended the commit to remove that line.